### PR TITLE
cinc-auditor: init at 5.21.29

### LINKED ISSUE #226772: Package request: cinc-auditor
**Project description**

Cinc Auditor is the Community build of Inspec by Chef.  Inspec is an open-source testing framework for infrastructure
  with a human- and machine-readable language for specifying compliance,  security and policy requirements.

**Metadata**

* homepage URL: https://cinc.sh/
* source URL: https://gitlab.com/cinc-project/distribution/auditor
* license: Apache License, Version 2.0
* platforms: unix

### PR DESCRIPTION
**Pull request by dylanmtaylor**
###### Description of changes

This adds the application cinc-auditor and closes #226772.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/misc/cinc-auditor/default.nix [hunk 1]
```diff
@@ -0,0 +1,20 @@
+{ lib, ruby, bundlerApp, bundlerUpdateScript }:
+
+bundlerApp {
+  pname = "cinc-auditor-bin";
```
Comment (SuperSandro2000) [inline pkgs/tools/misc/cinc-auditor/default.nix:4]:
```suggestion
  pname = "cinc-auditor";
```

File: pkgs/tools/misc/cinc-auditor/default.nix [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+  gemdir = ./.;
+
+  inherit ruby;
+
+  exes = ["cinc-auditor"];
+
+  passthru.updateScript = bundlerUpdateScript "cinc-auditor-bin";
+
+  meta = with lib; {
+    description = "Cinc Auditor is the community build of Inspec by Chef. Inspec is an open-source testing framework for infrastructure with a human- and machine-readable language for specifying compliance, security and policy requirements";
```
Comment (SuperSandro2000) [inline pkgs/tools/misc/cinc-auditor/default.nix:14]: description should not be that long. Anything lengthy should be written in longDescription

**Comment by SuperSandro2000** [comment]
Why is this packaged instead of chef auditor? I can't find any difference except branding.

File: pkgs/tools/misc/cinc-auditor/gemset.nix [hunk 3]
```diff
@@ -0,0 +1,2230 @@
...
+    };
+    version = "5.21.29";
+  };
+  inspec-core = {
+    dependencies = ["addressable" "chef-telemetry" "faraday" "faraday-follow_redirects" "hashie" "license-acceptance" "method_source" "mixlib-log" "multipart-post" "parallel" "parslet" "pry" "rspec" "rspec-its" "rubyzip" "semverse" "sslshake" "thor" "tomlrb" "train-core" "tty-prompt" "tty-table"];
+    groups = ["default"];
+    platforms = [];
+    source = {
+      remotes = ["https://packagecloud.io/cinc-project/stable"];
+      sha256 = "1irzd4nms4fpv5llkqsx8mlgyhgsibkzk293w31yixarg3jbnvd7";
```
Comment (SuperSandro2000) [inline pkgs/tools/misc/cinc-auditor/gemset.nix:1400]:
```suggestion
      sha256 = "0ik98jqr0js3dxl11jl0n1rqi82kcf47ynh393bn48y8cdcn80yd";
```

**Comment by dylanmtaylor** [comment]
> Why is this packaged instead of chef auditor? I can't find any difference except branding.

I have PRs for both. The major difference is licensing. https://cinc.sh/goals/

File: pkgs/tools/misc/cinc-auditor/default.nix (hunk 1 above)
Comment (dylanmtaylor) [reply pkgs/tools/misc/cinc-auditor/default.nix:4]: I think this is not what we want, their site says to use cinc-auditor-bin. https://cinc.sh/start/auditor/

This gem seems to provide the executable.

File: pkgs/tools/misc/cinc-auditor/default.nix (hunk 2 above)
Comment (dylanmtaylor) [reply pkgs/tools/misc/cinc-auditor/default.nix:14]: I will change the description.

File: pkgs/tools/misc/cinc-auditor/default.nix (hunk 1 above)
Comment (SuperSandro2000) [reply pkgs/tools/misc/cinc-auditor/default.nix:4]: Then we can name it like this. I first thought this is a binray suffix like it is used in arch.

**Comment by dylanmtaylor** [comment]
I am closing this as the Chef packages were merged in and I get the point that this is basically the same exact package but with a different name. The SHA checksum changing was also concerning from a maintenance perspective.

**Pull request closed without merging**